src/review-todo.js

```diff
@@ -0,0 +1,212 @@
+let isDate = require("date-fns/isDate");
+
+function todo(todoName = "New Todo") {
+	let title = todoName,
+		description = "",
+		urgency = 0,
+		isDone = false,
+		dueDate = todoDate(),
+		scheduleDate = todoDate(),
+		tags = tagList();
+
+	let todoObj = Object.create(todo.proto);
+	Object.assign(todoObj, {
+		title,
+		description,
+		isDone,
+		urgency,
+		dueDate,
+		scheduleDate,
+		tags,
+	});
+
+	return todoObj;
+}
+
+todo.proto = {};
+
+function project(projectName = "New Project") {
+	let proto = Object.assign(project.proto, { isChild });
+	let projectObj = Object.create(proto);
+
+	let title = projectName,
+		description = "",
+		favorite = false,
+		urgency = 0,
+		isDone = false,
+		dueDate = todoDate(),
+		scheduleDate = todoDate(),
+		tags = tagList(),
+		_todoList = createTodoList(projectObj),
+		todoList = Object.create(_todoList),
+		createTag = tagGenerator("project").bind(projectObj);
+
+	todoList.add = (proj) => {
+		if (
+			Object.getPrototypeOf(proj) === project.proto &&
+			projectObj.isChild(proj)
+		) {
+			throw "Looping projects";
+		}
+		_todoList.add(proj);
```

Comment:
What's this doing? Is it possible to pull your logic into a predicate function like `isOwnParent` or something appropriately named so it's easier to debug this `add` method in the future.